docs-generation/scripts/sync_frontend_translations.py

```
@@ -20,19 +30,80 @@
+        except Exception:
+            continue
+    print(f"⚠️  Warning: failed to read {locale_rel} from {repo} (tried {refs})")
+    return {}
+
+def load_frontend_translations_remote(lang):
+    """Fetch nodeDefs.json for a language from GitHub raw (master branch)"""
+    url = f"{REMOTE_BASE}/{lang}/nodeDefs.json"
     try:
-        with open(locale_file, 'r', encoding='utf-8') as f:
-            return json.load(f)
+        req = urllib.request.Request(url, headers={"User-Agent": "docs-generation"})
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)